lib/gooddata/lcm/actions/synchronize_clients.rb

```diff
@@ -98,6 +111,38 @@ def call(params)
           # Return results
           results
         end
+
+        def remove_old_master_workspace(params, segment_id, master_projects, number_of_deleted_projects)
+          removal_master_project_ids = []
+          latest_project = master_projects.pop
```

Comment:
The master_projects should always contains the latest_project, do not pop and push it back. Error can happen in between (cannot push the latest_project back to master_projects).
We should change master_projects, always keep it consistent, just remove a project from it when the project is deleted.
```
need_to_delete_projects.foreach { 
delete old_project_id
master_projects.remove(old_project_id) 
```

Reply:
updated it